lib/fibonacci.rb

```diff
@@ -1,8 +1,25 @@
 # Improved Fibonacci
 
-# Time Complexity - ?
-# Space Complexity - ?  (should be O(n))
+# Time Complexity - O(n)
+# Space Complexity - O(n)
 # Hint, you may want a recursive helper method
 def fibonacci(n)
```

Comment:
This works, but you are keeping the list of all the fibonacci numbers for the whole length of the recursive calls.  Instead you only need to keep the last 2 fibonacci numbers.